src/fparser/tests/test_splitline.py

```diff
@@ -0,0 +1,139 @@
+#      contributors may be used to endorse or promote products derived from
+#      this software without specific prior written permission.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+# DAMAGE.
+
+# Original author: Pearu Peterson <pearu@cens.ioc.ee>
+# First version created: May 2006
+
+"""
+Test parsing single Fortran lines.
+
+"""
+
+from fparser.splitline import splitparen, splitquote, string_replace_map
+
+
+def test_splitparen():
+    ''' Unit tests for splitparen function.'''
+    assert splitparen('abc') == ['abc']
+    assert splitparen('abc(1)') == ['abc', '(1)']
+    assert splitparen('abc(1) xyz') == ['abc', '(1)', ' xyz']
+    assert splitparen('a(b) = b(x,y(1)) b((a))') == \
+        ['a', '(b)', ' = b', '(x,y(1))', ' b', '((a))']
+    # pylint: disable=anomalous-backslash-in-string
+    assert splitparen('a(b) = b(x,y(1)) b\((a)\)') == \
+        ['a', '(b)', ' = b', '(x,y(1))', ' b\\(', '(a)', '\\)']
+    # pylint: enable=anomalous-backslash-in-string
+    assert splitparen('abc[1]') == ['abc', '[1]']
+    assert splitparen('abc[1,2,3]') == ['abc', '[1,2,3]']
+    assert splitparen('a[b] = b[x,y(1)] b((a))') == \
+        ['a', '[b]', ' = b', '[x,y(1)]', ' b', '((a))']
+    # pylint: disable=anomalous-backslash-in-string
+    assert splitparen('a[b] = b[x,y(1)] b\((a)\)') == \
+        ['a', '[b]', ' = b', '[x,y(1)]', ' b\\(', '(a)', '\\)']
+    # pylint: enable=anomalous-backslash-in-string
+    assert splitparen('integer a(3) = (/"a", "b", "c"/)') == \
+        ['integer a', '(3)', ' = ', '(/"a", "b", "c"/)']
+    assert splitparen(
+        'character(len=40) :: a(3) = (/"a[),", ",b,[(", "c,][)("/)') == \
+        ['character', '(len=40)', ' :: a', '(3)', ' = ',
+         '(/"a[),", ",b,[(", "c,][)("/)']
+    assert splitparen('integer a(3) = ["a", "b", "c"]') == \
+        ['integer a', '(3)', ' = ', '["a", "b", "c"]']
+    assert splitparen(
+        'character(len=40) :: a(3) = ["a[),", ",b,[(", "c,][)("]') == \
+        ['character', '(len=40)', ' :: a', '(3)', ' = ',
+         '["a[),", ",b,[(", "c,][)("]']
+    # pylint: disable=anomalous-backslash-in-string
+    result = splitparen('a(1),b\\((2,3),c\\\((1)),c"("')
+    expected = ['a', '(1)', ',b\(', '(2,3)', ',c\\\\', '((1))', ',c"("']
```

Comment:
I can see that you've brought this test in unchanged from how it was before and therefore you might not know the answer to this, but... why does e.g. "c\\\" become "c\\\\" while "b\\" becomes "b\"? My Fortran knowledge does not stretch to this situation. Maybe yours or that of @rupertford does?

Reply:
It's not a Fortran, but a python issue - I've modified the tests somewhat to be easier to understand. The previous tests made use of the fact that python would not change a '\(' - this stays a two character strings. So e.g.: 'c\\\(' == 'c\\\\(' in python:
```
>>> print 'b\(' == 'b\\(', 'c\\\(' == 'c\\\\('
True True
```
Hope this is easier to understand now - the actual test case was not changed, but the way the string was written for the test case was improved.